### ISSUE TITLE
UI Errors: error getting application: applications.argoproj.io "xxxx" not found

### ISSUE THREAD

**Issue by karlschriek** [argoproj/argo-cd#12752]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Ever since the release in 2.5.0 of multiple source namespaces (which we started using immediately by the way!) I am running into errors `error getting application: applications.argoproj.io "xxxx" not found` errors on the UI. I listed one such error (which has already been fixed) here: https://github.com/argoproj/argo-cd/issues/11239, but they have been popping up all over the place.

To recap, the main reason for this is that from 2.5.0 onward, the unique path for an application is now `[source-namespace-name]:[application-name]` whereas previously it was only `[application-name]`. It is however possible to still reference applications with the short-hand `[application-name]`, which then just becomes `argocd:[application-name]`. For that reason, you will only see these errors if you actually activate this feature and deploy something from a different namespace.

**To Reproduce**

Roll out a Application into a namespace other than argocd (see https://github.com/argoproj/argo-cd/pull/9755 for instructions on how to enable this feature; it doesn't work out of the box). One example where I get this  is when navigate to an Application, and then click on the card for the Application itself.

The screenshots below show how to trigger this. You can see in the logs from argocd-server below that it is clearly attempting to find "cert-manager" in the "argocd" namespace (i.e. making use of the shorthand).

 I don't have an exhaustive list of ways the error occurs, but I do know that I run into them frequently. It would probably be more useful to search through the code and try to deal with all incorrect occurrences, rather than to try to reproduce each individual error.

**Expected behavior**

I would expect all UI interactions to correctly use `[source-namespace-name]:[application-name]` instead of `[application-name]`.

**Screenshots**
![image](https://user-images.githubusercontent.com/25316920/223398574-660611c7-3af8-4a65-a4a3-6951e65bb9ae.png)

![image](https://user-images.githubusercontent.com/25316920/223398089-ec0206f2-e99b-42b3-8ea5-ecdedafaf191.png)

**Version**

```shell
v2.6.2+6e02f8b
```

**Logs**

```
time="2023-03-07T10:37:38Z" level=info msg="received unary call /application.ApplicationService/ListLinks" grpc.method=ListLinks grpc.request.claims="{...}" grpc.request.content="name:\"cert-manager\" " grpc.service=application.ApplicationService grpc.start_time="2023-03-07T10:37:38Z" span.kind=server system=grpc
time="2023-03-07T10:37:38Z" level=error msg="failed to get application, error=applications.argoproj.io \"cert-manager\" not found" application=cert-manager ns=argocd
time="2023-03-07T10:37:38Z" level=error msg="finished unary call with code Unknown" error="error getting application" grpc.code=Unknown grpc.method=ListLinks grpc.service=application.ApplicationService grpc.start_time="2023-03-07T10:37:38Z" grpc.time_ms=11.268 span.kind=server system=grpc
tim
```

**Comment by asoehnlein** [comment]
We are experiencing the exact same error ever since v2.5.0. 
Now running v2.6.5 it is still not fixed.
In addition it is also not possible to edit the Manifest since the application is "not found" due to error.

![argocd_error](https://user-images.githubusercontent.com/60323614/225598925-93f9be3b-2887-4191-8dec-3dc3fbddc345.png)

**Comment by crenshaw-dev** [comment]
This should be fixed in recent versions. If anyone encounters it again, ping me, and I'll reopen.

**Comment by tsouza-squid** [comment]
> This should be fixed in recent versions. If anyone encounters it again, ping me, and I'll reopen.

Is this supposed to be fixed in `2.9.3`?
